OneCommunityGlobal/HighestGoodNetworkApp
# Tim fix task info changes modal close bug

# Description
(PRIORITY LOW) Tim: Fix Task Info Changes modal (Dashboard > Tasks > Click red bell icon) to close properly. Clicking the close button or Mark As Read button does not fully close the modal. The modal header remains on screen. In this state the close button no longer has any effect. A page refresh is required to return to the Dashboard. 


## Related PRS (if any):
N/A

## Main changes explained:
- Simple fix: the state setter controlling modal on/off did not receive the latest state directly from the `showTaskNotificationModal` state value. The state setter function has been updated to ensure the latest state is used.
- A `key` prop has been added to the modal map function to remove a key prop bug.

## How to test:
- `git checkout Tim_fix_task_info_changes_modal_close_bug` to check into current branch
- `npm install` to ensure your modules are up to date (no new modules are added in this PR)
- `npm run start:local` to run this PR locally
- log in as an admin user to access the full list of tasks
- under **Dashboard > Team Member Tasks**, find a task  with a red bell icon and click to open the **Task Info Changes** modal.
   - Note: the task should preferably be one of your own tasks so you also have access to the **Mark as Read** button. If you don't have a task of your own you can create one as an admin. You can also temporarily remove the logic in **TaskDifferenceModal.jsx** that only allows the task owner access to **Mark as Read** (see below)

<img width="721" alt="markasread" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/be6b9ca9-8103-4eab-82fd-9e93efa3f7ce">

- click the `x` icon to close the modal. The modal should completely close out and you should have full access to the Dashboard again.
- reopen the modal and click the `Mark as Read` button to test the same. (Note: this will clear the bell icon, so test this after testing the regular close button).
   - Note: you can generate a new red bell icon for a task you own by changing the name

## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/90c0a836-4220-4b05-a839-56a99f4d0440

## Notes:
N/A
